From dbc0e98b62ca2961cd83ddf9fb2ad4ec80f941fe Mon Sep 17 00:00:00 2001
Message-Id: <dbc0e98b62ca2961cd83ddf9fb2ad4ec80f941fe@dist-git>
From: Andrea Bolognani <abologna@redhat.com>
Date: Tue, 4 Apr 2017 14:59:28 +0200
Subject: [PATCH] tests: Initialize basic capabilities properly

The capabilities used in test cases should match those used
during normal operation for the tests to make any sense.

This results in the generated command line for a few test
cases (most notably non-x86 test cases that were wrongly
assuming they could use -no-acpi) changing.

(cherry picked from commit 1cf3e52abb0ffbfe525a67d0d6379c45b51cd643)

Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1429509

Signed-off-by: Andrea Bolognani <abologna@redhat.com>
---
 src/qemu/qemu_capabilities.c                                          | 2 +-
 src/qemu/qemu_capabilities.h                                          | 2 ++
 tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-host.args             | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v2.args               | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v3.args               | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-balloon-ccw-deflate.args          | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-console-sclp.args                 | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-console-virtio-ccw.args           | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-console-virtio-s390.args          | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-features.args            | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-zEC12.args               | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw-many.args         | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw.args              | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-s390.args             | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-scsi-ccw.args         | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-fs9p-ccw.args                     | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-hostdev-scsi-vhost-scsi-ccw.args  | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-hugepages-numa.args               | 1 +
 tests/qemuxml2argvdata/qemuxml2argv-iothreads-disk-virtio-ccw.args    | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-iothreads-virtio-scsi-ccw.args    | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-cap.args   | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-caps.args  | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-cap.args    | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-caps.args   | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-cap.args   | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-caps.args  | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-cap.args    | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-caps.args   | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none-caps.args    | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none.args         | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-memory-hotplug-ppc64-nonuma.args  | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-net-virtio-ccw.args               | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-net-virtio-s390.args              | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-ppc-dtb.args                      | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-ppce500-serial.args               | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-basic.args                | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-compat.args           | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-exact.args            | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-le.args               | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-nvram.args                | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-missing.args        | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-no-address.args     | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-default.args          | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-kbd.args              | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-multi.args            | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-vio-user-assigned.args    | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-pseries-vio.args                  | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-q35-virt-manager-basic.args       | 1 +
 .../qemuxml2argv-s390-allow-bogus-usb-controller.args                 | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-none.args    | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-s390-panic-missing.args           | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-s390-panic-no-address.args        | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-virtio-rng-ccw.args               | 1 -
 tests/qemuxml2argvdata/qemuxml2argv-watchdog-diag288.args             | 1 -
 tests/qemuxml2argvtest.c                                              | 4 ++--
 55 files changed, 7 insertions(+), 53 deletions(-)

diff --git a/src/qemu/qemu_capabilities.c b/src/qemu/qemu_capabilities.c
index b568a0f5e..9d0a29732 100644
--- a/src/qemu/qemu_capabilities.c
+++ b/src/qemu/qemu_capabilities.c
@@ -4302,7 +4302,7 @@ virQEMUCapsInitQMPArch(virQEMUCapsPtr qemuCaps,
  *
  * Initialize @qemuCaps with basic architecture-dependent capabilities.
  */
-static void
+void
 virQEMUCapsInitQMPBasicArch(virQEMUCapsPtr qemuCaps)
 {
     /* ACPI/HPET/KVM PIT are x86 specific */
diff --git a/src/qemu/qemu_capabilities.h b/src/qemu/qemu_capabilities.h
index d44682f2a..24e2f38ee 100644
--- a/src/qemu/qemu_capabilities.h
+++ b/src/qemu/qemu_capabilities.h
@@ -510,6 +510,8 @@ int virQEMUCapsParseHelpStr(const char *qemu,
                             const char *qmperr);
 /* Only for use by test suite */
 int virQEMUCapsParseDeviceStr(virQEMUCapsPtr qemuCaps, const char *str);
+/* Only for use by test suite */
+void virQEMUCapsInitQMPBasicArch(virQEMUCapsPtr qemuCaps);
 
 VIR_ENUM_DECL(virQEMUCaps);
 
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-host.args b/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-host.args
index 23276ad71..5fdaa08ef 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-host.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-host.args
@@ -15,5 +15,4 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-aarch64test/monitor.sock,server,nowait \
--no-acpi \
 -boot c
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v2.args b/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v2.args
index f79c1c60e..2caf1b92c 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v2.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v2.args
@@ -15,5 +15,4 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-aarch64test/monitor.sock,server,nowait \
--no-acpi \
 -boot c
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v3.args b/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v3.args
index fce9308b0..3ae8e5f86 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v3.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-aarch64-gic-v3.args
@@ -15,5 +15,4 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-aarch64test/monitor.sock,server,nowait \
--no-acpi \
 -boot c
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-balloon-ccw-deflate.args b/tests/qemuxml2argvdata/qemuxml2argv-balloon-ccw-deflate.args
index 85650717e..c8bb7260f 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-balloon-ccw-deflate.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-balloon-ccw-deflate.args
@@ -15,6 +15,5 @@ QEMU_AUDIO_DRV=none \
 -nodefconfig \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -device virtio-balloon-ccw,id=balloon0,devno=fe.0.000a,deflate-on-oom=on
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-console-sclp.args b/tests/qemuxml2argvdata/qemuxml2argv-console-sclp.args
index dc9f7afeb..17f60550c 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-console-sclp.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-console-sclp.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-s390,drive=drive-virtio-disk0,id=virtio-disk0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-console-virtio-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-console-virtio-ccw.args
index 4fe0d6d39..73868d436 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-console-virtio-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-console-virtio-ccw.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -device virtio-serial-ccw,id=virtio-serial0,devno=fe.0.0001 \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-console-virtio-s390.args b/tests/qemuxml2argvdata/qemuxml2argv-console-virtio-s390.args
index 6e0e11e8a..06ef1b76c 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-console-virtio-s390.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-console-virtio-s390.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -device virtio-serial-s390,id=virtio-serial0 \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-s390,drive=drive-virtio-disk0,id=virtio-disk0,bootindex=1 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-features.args b/tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-features.args
index 07abc93b0..0b4ce4c68 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-features.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-features.args
@@ -15,5 +15,4 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-guest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-zEC12.args b/tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-zEC12.args
index 4c95d6ac6..48850b8ca 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-zEC12.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-cpu-s390-zEC12.args
@@ -15,5 +15,4 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-guest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw-many.args b/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw-many.args
index 5f6b08733..8290cd652 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw-many.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw-many.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=0.0.0007,drive=drive-virtio-disk0,id=virtio-disk0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw.args
index 45f51a8d1..2f77a2955 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-ccw.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-s390.args b/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-s390.args
index 30aa47a1d..45c465912 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-s390.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-s390.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-s390,drive=drive-virtio-disk0,id=virtio-disk0
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-scsi-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-scsi-ccw.args
index 3b352bf56..522fd9954 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-scsi-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-disk-virtio-scsi-ccw.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -device virtio-scsi-ccw,id=scsi0,devno=fe.0.0001 \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-fs9p-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-fs9p-ccw.args
index d4bf19caa..6d4f65e0c 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-fs9p-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-fs9p-ccw.args
@@ -15,7 +15,6 @@ QEMU_AUDIO_DRV=none \
 -nodefconfig \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-hostdev-scsi-vhost-scsi-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-hostdev-scsi-vhost-scsi-ccw.args
index 199adfa5d..03de632ec 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-hostdev-scsi-vhost-scsi-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-hostdev-scsi-vhost-scsi-ccw.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest2/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -device virtio-scsi-ccw,id=scsi0,devno=fe.0.0001 \
 -drive file=/dev/HostVG/QEMUGuest2,format=raw,if=none,id=drive-virtio-disk0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-hugepages-numa.args b/tests/qemuxml2argvdata/qemuxml2argv-hugepages-numa.args
index 26e9496cf..1b2a52bcc 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-hugepages-numa.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-hugepages-numa.args
@@ -22,6 +22,7 @@ host-nodes=1-3,policy=bind \
 -monitor unix:/tmp/lib/domain--1-fedora/monitor.sock,server,nowait \
 -rtc base=utc,driftfix=slew \
 -no-kvm-pit-reinjection \
+-no-hpet \
 -global PIIX4_PM.disable_s3=1 \
 -global PIIX4_PM.disable_s4=1 \
 -boot c \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-iothreads-disk-virtio-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-iothreads-disk-virtio-ccw.args
index 5e001de06..ba4e699ee 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-iothreads-disk-virtio-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-iothreads-disk-virtio-ccw.args
@@ -16,7 +16,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,iothread=iothread1,devno=fe.0.0000,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-iothreads-virtio-scsi-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-iothreads-virtio-scsi-ccw.args
index 1ebb6d4a8..e87912b96 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-iothreads-virtio-scsi-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-iothreads-virtio-scsi-ccw.args
@@ -16,7 +16,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -device virtio-scsi-ccw,iothread=iothread2,id=scsi0,devno=fe.0.0001 \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-cap.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-cap.args
index 33a53a0fa..c718084db 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-cap.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-cap.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-caps.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-caps.args
index 33a53a0fa..c718084db 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-caps.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-off-caps.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-cap.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-cap.args
index 9536cce61..c6eaa244b 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-cap.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-cap.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-caps.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-caps.args
index 9536cce61..c6eaa244b 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-caps.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-aeskeywrap-on-caps.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-cap.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-cap.args
index 9afa7c212..434f5061e 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-cap.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-cap.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-caps.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-caps.args
index 9afa7c212..434f5061e 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-caps.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-off-caps.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-cap.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-cap.args
index e73ed675f..24e2035a3 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-cap.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-cap.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-caps.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-caps.args
index e73ed675f..24e2035a3 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-caps.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-deakeywrap-on-caps.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none-caps.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none-caps.args
index 19d2c3520..acde08a4b 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none-caps.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none-caps.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none.args b/tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none.args
index 19d2c3520..acde08a4b 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-machine-keywrap-none.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-memory-hotplug-ppc64-nonuma.args b/tests/qemuxml2argvdata/qemuxml2argv-memory-hotplug-ppc64-nonuma.args
index 781f30db0..88ed6b615 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-memory-hotplug-ppc64-nonuma.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-memory-hotplug-ppc64-nonuma.args
@@ -18,7 +18,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -kernel /media/ram/uImage \
 -initrd /media/ram/ramdisk \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-net-virtio-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-net-virtio-ccw.args
index 86b6adde0..8a0598a9a 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-net-virtio-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-net-virtio-ccw.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -device virtio-net-ccw,vlan=0,id=net0,mac=00:11:22:33:44:55,devno=fe.0.0001 \
 -net user,vlan=0,name=hostnet0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-net-virtio-s390.args b/tests/qemuxml2argvdata/qemuxml2argv-net-virtio-s390.args
index 46ccbc90b..ef31b791a 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-net-virtio-s390.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-net-virtio-s390.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -device virtio-net-s390,vlan=0,id=net0,mac=00:11:22:33:44:55 \
 -net user,vlan=0,name=hostnet0
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-ppc-dtb.args b/tests/qemuxml2argvdata/qemuxml2argv-ppc-dtb.args
index 58c1526cd..10ddca0dc 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-ppc-dtb.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-ppc-dtb.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -kernel /media/ram/uImage \
 -initrd /media/ram/ramdisk \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-ppce500-serial.args b/tests/qemuxml2argvdata/qemuxml2argv-ppce500-serial.args
index d62383602..87f4ee5e4 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-ppce500-serial.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-ppce500-serial.args
@@ -16,7 +16,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -kernel /media/ram/uImage \
 -initrd /media/ram/ramdisk \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-basic.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-basic.args
index 2839dea62..97a7057ba 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-basic.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-basic.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -usb \
 -chardev pty,id=charserial0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-compat.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-compat.args
index 06595a2aa..7740e2f5a 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-compat.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-compat.args
@@ -18,7 +18,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -usb \
 -chardev pty,id=charserial0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-exact.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-exact.args
index 803c1aa2b..d2c99a7fa 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-exact.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-exact.args
@@ -18,7 +18,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -usb \
 -chardev pty,id=charserial0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-le.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-le.args
index 2839dea62..97a7057ba 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-le.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-cpu-le.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -usb \
 -chardev pty,id=charserial0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-nvram.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-nvram.args
index 89e0e56a2..904b9518e 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-nvram.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-nvram.args
@@ -14,7 +14,6 @@ QEMU_AUDIO_DRV=none \
 -nographic \
 -nodefaults \
 -monitor unix:/tmp/lib/domain--1-QEMUGuest1/monitor.sock,server,nowait \
--no-acpi \
 -boot c \
 -usb \
 -global spapr-nvram.reg=0x4000
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-missing.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-missing.args
index 2839dea62..97a7057ba 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-missing.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-missing.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -usb \
 -chardev pty,id=charserial0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-no-address.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-no-address.args
index 2839dea62..97a7057ba 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-no-address.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-panic-no-address.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -usb \
 -chardev pty,id=charserial0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-default.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-default.args
index 56e3552a8..a92b1e01b 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-default.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-default.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -device pci-ohci,id=usb,bus=pci.0,addr=0x1 \
 -chardev pty,id=charserial0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-kbd.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-kbd.args
index 1a672c62a..caaccdbb8 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-kbd.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-kbd.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -device pci-ohci,id=usb,bus=pci.0,addr=0x1 \
 -chardev pty,id=charserial0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-multi.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-multi.args
index 727389102..b9bd905a5 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-multi.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-usb-multi.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -device piix3-usb-uhci,id=usb,bus=pci.0,addr=0x1 \
 -device pci-ohci,id=usb1,bus=pci.0,addr=0x2 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-vio-user-assigned.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-vio-user-assigned.args
index 5754d3c9a..63cf3c183 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-vio-user-assigned.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-vio-user-assigned.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -device spapr-vscsi,id=scsi0,reg=0x2000 \
 -device spapr-vscsi,id=scsi1,reg=0x30000000 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-pseries-vio.args b/tests/qemuxml2argvdata/qemuxml2argv-pseries-vio.args
index 43381dd2e..0294067bc 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-pseries-vio.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-pseries-vio.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -device spapr-vscsi,id=scsi0,reg=0x2000 \
 -device spapr-vscsi,id=scsi1,reg=0x3000 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-q35-virt-manager-basic.args b/tests/qemuxml2argvdata/qemuxml2argv-q35-virt-manager-basic.args
index e139e5237..91598e9d8 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-q35-virt-manager-basic.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-q35-virt-manager-basic.args
@@ -15,6 +15,7 @@ QEMU_AUDIO_DRV=spice \
 -monitor unix:/tmp/lib/domain--1-virt-manager-basic/monitor.sock,server,nowait \
 -rtc base=utc,driftfix=slew \
 -no-kvm-pit-reinjection \
+-no-hpet \
 -global ICH9-LPC.disable_s3=1 \
 -global ICH9-LPC.disable_s4=1 \
 -boot c \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-controller.args b/tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-controller.args
index a1c1c85af..8abe0df98 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-controller.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-controller.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-test/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -device virtio-serial-s390,id=virtio-serial0 \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-s390,drive=drive-virtio-disk0,id=virtio-disk0,bootindex=1 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-none.args b/tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-none.args
index a1c1c85af..8abe0df98 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-none.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-s390-allow-bogus-usb-none.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-test/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -device virtio-serial-s390,id=virtio-serial0 \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-s390,drive=drive-virtio-disk0,id=virtio-disk0,bootindex=1 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-s390-panic-missing.args b/tests/qemuxml2argvdata/qemuxml2argv-s390-panic-missing.args
index a18b5a5a7..32678145a 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-s390-panic-missing.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-s390-panic-missing.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -device virtio-serial-ccw,id=virtio-serial0,devno=fe.0.0000 \
 -chardev pty,id=charconsole0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-s390-panic-no-address.args b/tests/qemuxml2argvdata/qemuxml2argv-s390-panic-no-address.args
index e49c8df83..fa8fbbc4a 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-s390-panic-no-address.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-s390-panic-no-address.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -boot c \
 -device virtio-serial-ccw,id=virtio-serial0,devno=fe.0.0000 \
 -chardev pty,id=charconsole0 \
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-virtio-rng-ccw.args b/tests/qemuxml2argvdata/qemuxml2argv-virtio-rng-ccw.args
index dec49e1eb..a73bce85d 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-virtio-rng-ccw.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-virtio-rng-ccw.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -device virtio-serial-ccw,id=virtio-serial0,devno=fe.0.0001 \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-ccw,devno=fe.0.0000,drive=drive-virtio-disk0,\
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-watchdog-diag288.args b/tests/qemuxml2argvdata/qemuxml2argv-watchdog-diag288.args
index 2997c1241..95f915c2b 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-watchdog-diag288.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-watchdog-diag288.args
@@ -17,7 +17,6 @@ QEMU_AUDIO_DRV=none \
 -chardev socket,id=charmonitor,path=/tmp/lib/domain--1-QEMUGuest1/monitor.sock,\
 server,nowait \
 -mon chardev=charmonitor,id=monitor,mode=readline \
--no-acpi \
 -drive file=/dev/HostVG/QEMUGuest1,format=raw,if=none,id=drive-virtio-disk0 \
 -device virtio-blk-s390,drive=drive-virtio-disk0,id=virtio-disk0,bootindex=1 \
 -device diag288,id=watchdog0 \
diff --git a/tests/qemuxml2argvtest.c b/tests/qemuxml2argvtest.c
index 41461eadc..731b793c4 100644
--- a/tests/qemuxml2argvtest.c
+++ b/tests/qemuxml2argvtest.c
@@ -360,8 +360,6 @@ testInitQEMUCaps(struct testInfo *info,
     if (!(info->qemuCaps = virQEMUCapsNew()))
         goto cleanup;
 
-    virQEMUCapsSet(info->qemuCaps, QEMU_CAPS_NO_ACPI);
-
     if (testQemuCapsSetGIC(info->qemuCaps, gic) < 0)
         goto cleanup;
 
@@ -381,6 +379,8 @@ testUpdateQEMUCaps(const struct testInfo *info,
 
     virQEMUCapsSetArch(info->qemuCaps, vm->def->os.arch);
 
+    virQEMUCapsInitQMPBasicArch(info->qemuCaps);
+
     /* We need to pretend QEMU 2.0.0 is in use so that pSeries guests
      * will get the correct alias assigned to their buses.
      * See virQEMUCapsHasPCIMultiBus() */
-- 
2.12.2

